### ISSUE TITLE
libjpeg-01: Compilation with anything beyond -O0 results in unusable JavaScript code

### ISSUE THREAD

**Issue by cee-dee** [cee-dee/emscripten-tests#3]
Follow the build-instructions in emscripten-tests/libjpeg-01/README.txt. The resulting JavaScript will be of a size of about 4 MB, because it is built using -O0, without using the closure-compiler.

For increasing execution-speed and for reducing the size, I'd like to compile with at least -O2, better with -Oz.

So far I've found no way to this. It tried just calling emcc with -O2/-Oz instead of -O0. I additionally tried to compile libjpeg with emmake make -e CPPFLAGS=-Oz, before compiling with emcc -Oz, but that did not work either. I get exceptions like this all the time:
# 

uncaught exception: abort(0) at jsStackTrace@file:///home/user/emscripten/jpeg-components/ipl-wrapper/emscripten-tests/libjpeg-01/src/DecodeImage.js:1:21746
stackTrace@file:///home/user/emscripten/jpeg-components/ipl-wrapper/emscripten-tests/libjpeg-01/src/DecodeImage.js:1:21929
abort@file:///home/user/emscripten/jpeg-components/ipl-wrapper/emscripten-tests/libjpeg-01/src/DecodeImage.js:15:35454
nullFunc_vii@file:///home/user/emscripten/jpeg-components/ipl-wrapper/emscripten-tests/libjpeg-01/src/DecodeImage.js:1:192767
im@file:///home/user/emscripten/jpeg-components/ipl-wrapper/emscripten-tests/libjpeg-01/src/DecodeImage.js:8:33536
he@file:///home/user/emscripten/jpeg-components/ipl-wrapper/emscripten-tests/libjpeg-01/src/DecodeImage.js:5:29651
$d@file:///home/user/emscripten/jpeg-components/ipl-wrapper/emscripten-tests/libjpeg-01/src/DecodeImage.js:5:11423
Qd@file:///home/user/emscripten/jpeg-components/ipl-wrapper/emscripten-tests/libjpeg-01/src/DecodeImage.js:5:4413
Pd@file:///home/user/emscripten/jpeg-components/ipl-wrapper/emscripten-tests/libjpeg-01/src/DecodeImage.js:5:3870
ud@file:///home/user/emscripten/jpeg-components/ipl-wrapper/emscripten-tests/libjpeg-01/src/DecodeImage.js:5:868
Id@file:///home/user/emscripten/jpeg-components/ipl-wrapper/emscripten-tests/libjpeg-01/src/DecodeImage.js:5:2636
pl@file:///home/user/emscripten/jpeg-components/ipl-wrapper/emscripten-tests/libjpeg-01/src/DecodeImage.js:8:31500
dynCall_viiii_5@file:///home/user/emscripten/jpeg-components/ipl-wrapper/emscripten-tests/libjpeg-01/src/DecodeImage.js line 1 > Function:2:12
DecodeImage$initialize@file:///home/user/emscripten/jpeg-components/ipl-wrapper/emscripten-tests/libjpeg-01/src/DecodeImage.js line 1 > Function:8:1
processJpgArray@file:///home/user/emscripten/jpeg-components/ipl-wrapper/emscripten-tests/libjpeg-01/src/loadJpgs.html:28:9
loadJpgFile/fileReader.onload@file:///home/user/emscripten/jpeg-components/ipl-wrapper/emscripten-tests/libjpeg-01/src/loadJpgs.html:99:11
 <unbekannt>
"Invalid function pointer '0' called with signature 'vii'. Perhaps this is an invalid value (e.g. caused by calling a virtual method on a NULL pointer)? Or calling a function with an incorrect type, which will fail? (it is worth building your source files with -Werror (warnings are errors), as warnings can indicate undefined behavior which can cause this)" DecodeImage.js:1:2539
"This pointer might make sense in another type signature: vi: 0  viii: 0  v: 0  viiii: 0  viiiii: 0  viiiiii: 0  viiiiiii: 0  ii: 0  iii: 0  i: 0  iiii: 0  iiiii: 0  iiiiii: 0  iiiiiii: 0  " DecodeImage.js:1:2539
0 DecodeImage.js:1:2460
#0 

Any ideas how to fix this?

**Comment by waywardmonkeys** [comment]
I believe this is a bug in your code which I described in a post here:

```
https://groups.google.com/forum/#!topic/emscripten-discuss/m-U1c2h62Jg
```

**Comment by cee-dee** [comment]
@waywardmonkeys: Right, that's the issue! I just pushed the fix https://github.com/cee-dee/emscripten-tests/commit/d81010fd7e3dd39aa23fea59b8c1f813726e3c68

Work's as you suggested, many many thanks for that! I'll post more information within the Google group.
